src/lib/error-handler/index.ts

```diff
@@ -14,18 +51,14 @@ export const errorHandler = function errorHandler(
   next: NextFunction
 ): ExResponse | void {
   if (err instanceof ValidateError) {
-    logger.debug(`Handled Validation Error for ${req.path}:`, err.fields)
-    const response: ValidateErrorJSON = {
-      message: 'Validation failed',
-      details: err?.fields,
-    }
-    return res.status(422).json(response)
+    logger.error(`Handled Validation Error for ${req.path}:`, err.fields)
+
+    return res.send(err)
   }
   if (err instanceof Error) {
-    logger.warn('Unexpected error thrown in handler: %s', err.message)
-    return res.status(500).json({
-      message: 'Internal Server Error',
-    })
+    logger.error('Unexpected error thrown in handler: %s', err.message)
+
+    return res.status(500).json(err)
```

Comment:
I think there needs to be code in the handler to handle each error type and return the correct code. At the moment the code is always `500` when there's an error. The `  @Response<BadRequest>(400)` type decorators don't seem to pass a code through?

Reply:
for example, getting an attachment with a bad id should `404` but it `500`s
![image](https://user-images.githubusercontent.com/40722025/225595130-ac77ba8b-2365-4fe0-9c48-2abeae6338fc.png)